include/nuttx/list.h

```diff
@@ -274,6 +274,12 @@
       &entry->member != (list); entry = temp, \
       temp = list_container_of(temp->member.next, type, member))
 
+/* Iterate from a given entry node */
+
+#define list_for_every_entry_from(list, cur, type, member) \
```

Comment:
2 optimization points in further:
1. rename all **for_every** to **for_each**
2. remove type parameter, instead of typeof